### PR TITLE
refactor: new digits length rule for 2FA setup

### PR DESCRIPTION
**Pull request by Highjhacker**
## Summary

https://app.clickup.com/t/hq4nng

## Checklist

- [ ] Documentation _(if necessary)_
- [ ] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

File: src/Components/TwoFactorAuthenticationForm.php [hunk 1]
```diff
@@ -51,7 +51,7 @@ public function render(): View
     public function enableTwoFactorAuthentication(): void
     {
         $this->validate([
-            'state.otp' => ['required', new OneTimePassword($this->state['two_factor_secret'])],
+            'state.otp' => ['required', new OneTimePassword($this->state['two_factor_secret']), 'digits_between:4,10'],
```
Comment (ItsANameToo) [inline src/Components/TwoFactorAuthenticationForm.php]: 6, otp is always 6

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

**Comment by ItsANameToo** [comment]
@Highjhacker have you tried installing this on MSQ?

**Comment by Highjhacker** [comment]
Na, tried it by modifying the file within `vendor/` and tested it from there and also ran the tests on `laravel-ui` and was all green, I guess something is wrong ? :trollface:

**Review by ItsANameToo — CHANGES_REQUESTED** [review]
this doesn't seem to do much 🤔 and when you enable it it will simply say the code does not work, nothing regarding to being limited to 6
image

File: src/Components/TwoFactorAuthenticationForm.php [hunk 2]
```diff
@@ -51,7 +51,7 @@ public function render(): View
     public function enableTwoFactorAuthentication(): void
     {
         $this->validate([
-            'state.otp' => ['required', new OneTimePassword($this->state['two_factor_secret'])],
+            'state.otp' => ['required', new OneTimePassword($this->state['two_factor_secret']), 'digits:6'],
```
Comment (ItsANameToo) [inline src/Components/TwoFactorAuthenticationForm.php]: switch order, `digits:6` before the `new OneTimePassword` rule

**Review by ItsANameToo — CHANGES_REQUESTED** [review]
- make the input field also take 6 digits max or people will still be able to fill in more than 6 digits as validation only hits when pressing the button
- above applies to all OTP fields, keep it consistent and limit them to 6 in the frontend already wherever they are used

**Comment by Highjhacker** [comment]
@ItsANameToo Applied the `max=6` on inputs but it doesn't seems to work at all 🤔 Also I don't think I can add much to the validation for the Login part, or at least haven't found how, checked a bit at `laravel-fortify` since most if from there but nothing.

**Comment by ItsANameToo** [comment]
@Highjhacker because passing `max` sets `maxlength` and that doesn't work on number inputs. This needs to be refactored: https://github.com/ArkEcosystem/laravel-ui/blob/master/resources/views/inputs/includes/input-field.blade.php#L25 and was the plan for a while. I don't think anything uses that property anymore but you would have to recheck that. If it's used then document it so it can be updated when the UI lib gets updated on the respective project

**Comment by ItsANameToo** [comment]
@Highjhacker sidenote, `max` will work for number inputs but that doesn't define the length but an actual number

**Comment by Highjhacker** [comment]
@ItsANameToo Used pattern but it's not like it's effectively preventing user to input more than 6 numbers, think we would need some JS for that, want me to have a look or it's fine as it is ? 🤔

**Comment by ItsANameToo** [comment]
nah if `max` doesn't cut it then so be it, backend only guard then

**Review by ItsANameToo — APPROVED** [review]

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**